File: _14889.java
import java.util.*;
public class _14889{
	static boolean[] check;
	static int person[][],n,dap=Integer.MAX_VALUE;
	static void calc(){
		int a=0,b=0;
		for(int i=1;i<=n;i++) {
			for(int j=1;j<=n;j++) {
				if(check[i]&&check[j]) a+=person[i][j];
				else if(!check[i]&&!check[j]) b+=person[i][j];
			}
		}
		dap = Math.min(dap, Math.abs(a-b));
	}
	static void solve(int count, int dep) {
		if(count==n/2) {
			calc();
		}
		else {
			for(int i=dep+1;i<=n;i++) {
				check[i]= true;
				solve(count+1,i);
				check[i]=false;
			}
		}
	}
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in); 
		n= sc.nextInt();
		check = new boolean[n+1];
		person = new int[n+1][n+1];
		for(int i=1;i<=n;i++) {
			for(int j=1;j<=n;j++) {
				person[i][j] = sc.nextInt();
			}
		}
			solve(0,0);
			System.out.println(dap);
	}

}

File: C++/14889.cpp
#include <stdio.h>
#include <functional>
#include <algorithm>
int map[20][20]; int n;
bool chk[20] = { 0 };
int result = INT32_MAX;
using namespace std;
void calc() {
	int a=0, b=0;
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			if (chk[i] && chk[j]) {
				a += map[i][j];
			}
			else if(!chk[i]&&!chk[j]){
				b += map[i][j];
			}
		}
	}
	result = __min(result, abs(a - b));
}
void solve(int ch, int dep) {
	if (n / 2 == ch) {
		calc();
	}
	else {
		for (int i = dep; i < n; i++) {
			chk[i] = true;
			solve(ch+1,i+1);
			chk[i] = false;
		}
	}
}
int main() {
	 scanf_s("%d", &n);
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			scanf_s("%d", &map[i][j]);
		}
	}
	solve(0,0);
	printf("%d", result);
}